### PR TITLE
made argparese in hf and bb to be importable

### PR DESCRIPTION
**Pull request by shot0829**
#### Checklist
- Have you updated the README (yes/no)?
- Have you updated the CHANGELOG (yes/no)?

### CODE REVIEW

**Comment by shot0829** [comment]
DO NOT MERGE!!
this PR is to show the difference of style of update.
i have not commit other codes needed.

File: scripts/hf_sim.py [hunk 1]
```diff
@@ -163,63 +169,132 @@ def random_seed():
         default=["1", "-1", "-1"],
     )
 
-    try:
-        args = parser.parse_args()
-    except SystemExit as e:
-        print(e, flush=True)
-        # invalid arguments or -h
-        comm.Abort()
+    args = parser.parse_args(cmd)
```
Comment (jpa198) [inline scripts/hf_sim.py:172]: Clever, leaving it as None results in default behaviour. Maybe add that to the docstring though?

**Comment by jpa198** [comment]
The diff is mostly showing the same content moved around, are there any differences aside from the top level content being in main function for each of them?

**Comment by shot0829** [comment]
> The diff is mostly showing the same content moved around, are there any differences aside from the top level content being in main function for each of them?

modulizing those functions means that it a variable has to passed around, when there is no forced class/definition of the structure of that variable, which may lead to future confusions.

File: scripts/cybershake/install_cybershake_fault.py [hunk 2]
```diff
@@ -64,6 +72,31 @@ def main():
...
+        if (any([ i == x for x in ['srun','python']]) or i[-3:] == '.py'):
+            command_parts.remove(i)
+            
+    for key in add_args:
+        command_parts.append("--" + key)
+        if add_args[key] is True:
+            continue
+        command_parts.append(str(add_args[key]))
+
+    return (list(map(str, command_parts)))
```
Comment (jpa198) [inline scripts/cybershake/install_cybershake_fault.py]: All elements of command_parts seem to be a string already?

File: scripts/cybershake/install_cybershake_fault.py [hunk 3]
```diff
@@ -223,6 +253,41 @@ def install_fault(
             vm_params_dict,
         )
 
+        # test if the params are accepted by steps HF and BB
+        sim_params = utils.load_sim_params(os.path.join(sim_dir, "sim_params.yaml"))
+        # check hf
+
+        # temporary change the script name to hf_sim, due to how error message are shown
+        main_script_name = sys.argv[0]
+        sys.argv[0] = "hf_simi.py"
```
Comment (jasonmotha) [inline scripts/cybershake/install_cybershake_fault.py:262]: hf_sim.py ??

File: scripts/submit_bb.py [hunk 4]
```diff
@@ -18,6 +18,19 @@
 default_wct = "00:30:00"
 
 
+def gen_command_template(params):
+    command_template_parameters = {
+        "outbin_dir": sim_struct.get_lf_outbin_dir(params.sim_dir),
```
Comment (jasonmotha) [inline scripts/submit_bb.py:23]: consistency. Probably best to use simulation_structure everywhere.

File: scripts/cybershake/install_cybershake_fault.py [hunk 5]
```diff
@@ -64,6 +73,27 @@ def main():
...
+def gen_args_cmd(
+    process: ProcessType, sim_dir, command_template, template_parameters, add_args={}
+):
+    # this function is adapted from generate_command from shared_workflow.shared_template
+    command_parts = []
+
+    command_parts = command_template.format(**template_parameters).split()
+    # remove srun, python, and *.py from the command
+    for i in list(command_parts):
+        if any([i == x for x in ["srun", "python"]]) or i[-3:] == ".py":
```
Comment (jpa198) [inline scripts/cybershake/install_cybershake_fault.py:85]: `if i in ["srun", "python"] or i.endswith(".py")`?

File: scripts/cybershake/install_cybershake_fault.py (hunk 3 above)
Comment (shot0829) [reply scripts/cybershake/install_cybershake_fault.py:262]: oops

File: scripts/cybershake/install_cybershake_fault.py (hunk 5 above)
Comment (jpa198) [reply scripts/cybershake/install_cybershake_fault.py]: Is there a reason for the list comprehension still being in there?

File: scripts/cybershake/install_cybershake_fault.py (hunk 2 above)
Comment (shot0829) [reply scripts/cybershake/install_cybershake_fault.py]: I think its better to keep it there, as it is not computational heavy (can only be so much args) and work as a last safety check if any unexpected values are in there (very unlikely but better be safe than sorry).

**Pull request merged**